src/rely/Test.re

```diff
@@ -4,8 +4,37 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  */;
+type matchers('ext) = {
```

Comment:
At some point I think it makes sense to extract these types to a public module that we expose directly via RelyAPI. Right now the record types are available but are defined on internal modules which is kind of weird. This wasn't obvious to me until I needed to add these new matchers. I don't think that change would be breaking, but I'm not 100% confident and this is equivalent to the status quo